src/features/create-dao/Steps/TreasuryForm.test.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+
+import { render, waitFor, screen, fireEvent } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { ZUIProvider } from '@zero-tech/zui/ZUIProvider';
+import { TreasuryForm,TreasuryFormProps } from './';
+import { CreateDAOFormContext } from "../";
+
+let onSubmit = jest.fn();
+
+const DEFAULT_PROPS: TreasuryFormProps = {
+	onClose: jest.fn(),
+};
+
+const DEFAULT_PROVIDER_VALUES = {
+	stepId: 'treasury',
+	title: 'Create DAO',
+	details: {
+		mediaType: undefined,
+		previewUrl: '',
+		name: '',
+		znaAddress: '',
+		description: '',
+	},
+	governance: {
+		votingProcess: '',
+		votingPeriod: '',
+		votingSystem: '',
+		daoTokenAddress: '',
+		votingThreshold: '',
+	},
+	treasury: {
+		gnosisSafe: '',
+	},
+	onStepUpdate: jest.fn(),
+	onTitleUpdate: jest.fn(),
+	onDetailsSubmit: jest.fn(),
+	onGovernanceSubmit: jest.fn(),
+	onTreasurySubmit: onSubmit,
+	onLaunchSubmit: jest.fn(),
+};
+
+describe('<TreasuryForm />', () => {
+	beforeEach(() => jest.resetAllMocks());
+
+
+	test('should correctly validate required Gnosis Safe field', async () => {
+		render (
+			<ZUIProvider>
+				<CreateDAOFormContext.Provider value={DEFAULT_PROVIDER_VALUES}>
+					<TreasuryForm {...DEFAULT_PROPS} />
+				</CreateDAOFormContext.Provider>
+			</ZUIProvider>
+		);
+
+		fireEvent.blur(screen.getByPlaceholderText(/Gnosis Safe/i));
+		fireEvent.click(
+			screen.getByRole('button', {
+				name: 'Next',
+			}),
+		);
+
+		await waitFor(() =>
+			expect(screen.getByText('Gnosis Safe is required')).not.toBe(
```

Comment:
Will also need to update this string to 'The gnosis safe field is required.'.

Reply:
Updated